s3-sink-connector/src/test/java/io/aiven/kafka/connect/s3/S3SinkTaskTest.java

```diff
@@ -689,17 +687,175 @@ void supportUnwrappedJsonEnvelopeForStructAndClassicJson() throws IOException {
+                "prefix-topic0-0-00000000000000000013" + compressionType.extension(),
+                "prefix-topic0-1-00000000000000000023" + compressionType.extension(),
+                "prefix-topic1-0-00000000000000000033" + compressionType.extension());
+        assertThat(expectedBlobs).allMatch(blobName -> testBucketAccessor.doesObjectExist(blobName));
+
+    }
+
+    @Test
+    void mutliPartUploadUsingKeyPartitioning() throws IOException {
+        final String compression = "none";
+        properties.put(S3SinkConfig.FILE_COMPRESSION_TYPE_CONFIG, compression);
+        properties.put(S3SinkConfig.FORMAT_OUTPUT_FIELDS_CONFIG, "value");
+        properties.put(S3SinkConfig.FORMAT_OUTPUT_ENVELOPE_CONFIG, "false");
+        properties.put(S3SinkConfig.FORMAT_OUTPUT_TYPE_CONFIG, "json");
+        properties.put(S3SinkConfig.AWS_S3_PREFIX_CONFIG, "prefix-");
+        // Compact/key 'mode' value only updated
+        properties.put(S3SinkConfig.FILE_NAME_TEMPLATE_CONFIG, "{{key}}-{{topic}}");
+
+        final S3SinkTask task = new S3SinkTask();
+        task.start(properties);
+        int timestamp = 1000;
+        int offset1 = 10;
+        int offset2 = 20;
+        int offset3 = 30;
+        final List<List<SinkRecord>> allRecords = new ArrayList<>();
+        for (int i = 0; i < 3; i++) {
+            allRecords.add(List.of(
+                    createRecordWithStructValueSchema("topic0", 0, "key0", "name0" + i, offset1++, timestamp++),
+                    createRecordWithStructValueSchema("topic0", 1, "key1", "name1" + i, offset2++, timestamp++),
+                    createRecordWithStructValueSchema("topic1", 0, "key2", "name2" + i, offset3++, timestamp++)));
+        }
+        final TopicPartition tp00 = new TopicPartition("topic0", 0);
+        final TopicPartition tp01 = new TopicPartition("topic0", 1);
+        final TopicPartition tp10 = new TopicPartition("topic1", 0);
+        final Collection<TopicPartition> tps = List.of(tp00, tp01, tp10);
+        task.open(tps);
+
+        allRecords.forEach(task::put);
+
+        final Map<TopicPartition, OffsetAndMetadata> offsets = new HashMap<>();
+        offsets.put(tp00, new OffsetAndMetadata(offset1));
+        offsets.put(tp01, new OffsetAndMetadata(offset2));
+        offsets.put(tp10, new OffsetAndMetadata(offset3));
+        task.flush(offsets);
+
+        final CompressionType compressionType = CompressionType.forName(compression);
+
+        List<String> expectedBlobs = Lists.newArrayList(
+                "prefix-topic0-0-00000000000000000012" + compressionType.extension(),
+                "prefix-topic0-1-00000000000000000022" + compressionType.extension(),
+                "prefix-topic1-0-00000000000000000032" + compressionType.extension());
+
+        assertThat(expectedBlobs).allMatch(blobName -> testBucketAccessor.doesObjectExist(blobName));
+
+        assertThat(testBucketAccessor.readLines("prefix-topic0-0-00000000000000000012", compression))
```

Comment:
As an FYI, the S3MockApi does not create the file names correctly for key, value